src/App.js

```diff
@@ -10,14 +10,44 @@ function App() {
   const [error, setError] = useState(null);
   const [isLoaded, setIsLoaded] = useState(false);
   const [city, setCity] = useState("New York City");
-  const [coordinates, setCoordinates] = useState({
-    lat: 40.7127,
-    lng: -74.0061,
-  });
+  const [coordinates, setCoordinates] = useState(null);
   const [results, setResults] = useState(null);
   const [generic, setGeneric] = useState("app");
   const [notfound, setFlag] = useState(false);
 
+  // get geolocation of the user
+  useEffect(() => {
+    if ("geolocation" in navigator) {
+      navigator.geolocation.getCurrentPosition(
+        (position) => {
+          setCoordinates({
+            lat: position.coords.latitude,
+            lng: position.coords.longitude,
+          });
+
+          // fetch city name based on coordinates
+          fetch(
+            `http://api.openweathermap.org/geo/1.0/reverse?lat=${position.coords.latitude}&lon=${position.coords.longitude}&appid=${process.env.REACT_APP_APIKEY}`
+          )
+            .then((response) => response.json())
+            .then((data) => {
+              setCity(data[0].name);
+            })
+            .catch((error) => {
+              console.error(
+                `Couldn't fetch city based on user's location: ${error}`
+              );
+            });
+        },
+        (error) => {
+          console.error(
```

Comment:
![image](https://user-images.githubusercontent.com/63879595/180961323-fbe8f77c-45d8-475f-a821-9fe2eed57270.png)
It is showing error even when the API is fetching the data.

Reply:
Some issue is there with fetching coordinates of location look into that